### PR TITLE
New features

### PR DESCRIPTION
**Pull request by Bobur-777**
### New features:
1. Added dark mode to show source code of current activity dialog.
2. Code highlight for dark theme (Dracula)
3.  Vector drawable viewer - It can only view vector drawables (not shapes etc.)
4. Added sort option for StringsViewer.
### Updates:
1. Changed Developer Tools activity's icons.
2. Changed DesignDrawer's icons.
3. Added icons to PopupMenu of DesignActivity.
4. Changed Property items icons
### Screenshoots:
image
image
...

### CODE REVIEW

**Comment by ilyassesalama** [comment]
Great changes as usual, but I think you could’ve separated them into different pull requests. Because you added major changes in many parts of the app.

**Comment by ilyassesalama** [comment]
> > Great changes as usual, but I think you could’ve separated them into different pull requests. Because you added major changes in many parts of the app.
> 
> Yes brother, Can you merge it? I don't know github because so many changes

Sure, but can you resolve the current conflicts.

File: app/src/main/java/com/besome/sketch/design/DesignActivity.java [hunk 1]
```diff
@@ -227,6 +233,8 @@ private void indicateCompileErrorOccurred(String error) {
                 startActivity(intent);
             }
         });
+        /* Set the text color to yellow */
+        snackbar.setActionTextColor(Color.YELLOW);
```
Comment (aikrq) [inline app/src/main/java/com/besome/sketch/design/DesignActivity.java:237]: why did you added this?

File: app/src/main/java/mod/bobur/BoburUtils.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package mod.bobur;
```
Comment (aikrq) [inline app/src/main/java/mod/bobur/BoburUtils.java:1]: So unnecessary. Class `ThemeUtils` already have method to check current device theme.

File: app/src/main/java/mod/hey/studios/code/SrcCodeEditor.java [hunk 3]
```diff
@@ -414,4 +421,8 @@ public static void showSwitchLanguageDialog(Activity activity, CodeEditor codeEd
                 .setNegativeButton(R.string.common_word_cancel, null)
                 .show();
     }
+    public static boolean isDarkModeEnabled(Context context) {
+        int nightModeFlags = context.getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
+        return nightModeFlags == Configuration.UI_MODE_NIGHT_YES;
+    }
```
Comment (aikrq) [inline app/src/main/java/mod/hey/studios/code/SrcCodeEditor.java:427]: Replace it with `ThemeUtils#isDarkThemeEnabled(Context)`.

File: app/src/main/java/mod/hilal/saif/activities/tools/AppSettings.java [hunk 4]
```diff
@@ -146,6 +148,12 @@ private void createToolsView(int icon, String title, String desc, LinearLayout t
         LibraryItemView item = new LibraryItemView(this);
         item.enabled.setVisibility(View.GONE);
         item.icon.setImageResource(icon);
+
+        TypedValue typedValue = new TypedValue();
+        getTheme().resolveAttribute(android.R.attr.colorPrimary, typedValue, true);
+        int colorPrimary = typedValue.data;
```
Comment (aikrq) [inline app/src/main/java/mod/hilal/saif/activities/tools/AppSettings.java:154]: What about to create another method in `ThemeUtils` to get theme color.

Maybe `ThemeUtils#getColorFromTheme`?

File: app/src/main/java/pro/sketchware/lib/highlighter/SyntaxScheme.java [hunk 5]
```diff
@@ -30,30 +42,40 @@ public SyntaxScheme(Pattern pattern, int color) {
...
+        arrayList.add(new SyntaxScheme(Pattern.compile(mXmlPattern[3]), getColor(SECONDARY_COLOR_LIGHT, SECONDARY_COLOR_DARK, isDarkMode)));
+        arrayList.add(new SyntaxScheme(Pattern.compile(mXmlPattern[1]), getColor(COMMENTS_COLOR_LIGHT, COMMENTS_COLOR_DARK, isDarkMode)));
+        arrayList.add(new SyntaxScheme(Pattern.compile(mJavaPattern[8]), getColor(QUOTES_COLOR_LIGHT, QUOTES_COLOR_DARK, isDarkMode)));
+
         return arrayList;
     }
 
+    private static int getColor(String lightColor, String darkColor, boolean isDarkMode) {
+        return Color.parseColor(isDarkMode ? darkColor : lightColor);
+    }
```
Comment (aikrq) [inline app/src/main/java/pro/sketchware/lib/highlighter/SyntaxScheme.java:77]: Maybe move to `ThemeUtils`?

**Review by aikrq — CHANGES_REQUESTED** [review]

**Comment by laymintu** [comment]
Can i get this version please??

File: app/src/main/java/com/besome/sketch/design/DesignActivity.java (hunk 1 above)
Comment (Bobur-777) [reply app/src/main/java/com/besome/sketch/design/DesignActivity.java:237]: I don't know 🥲, I think I am not added it

File: app/src/main/java/mod/bobur/BoburUtils.java (hunk 2 above)
Comment (Bobur-777) [reply app/src/main/java/mod/bobur/BoburUtils.java:1]: Yeah after 2-3 days I found ThemeUtils then I am not deleted my class

File: app/src/main/java/com/besome/sketch/design/DesignActivity.java (hunk 1 above)
Comment (aikrq) [reply app/src/main/java/com/besome/sketch/design/DesignActivity.java:237]: Remove it

**Comment by laymintu** [comment]
DO it ∆
I am still waiting 👀

File: app/src/main/java/mod/hilal/saif/activities/tools/AppSettings.java (hunk 4 above)
Comment (trindadedev13) [reply app/src/main/java/mod/hilal/saif/activities/tools/AppSettings.java:154]: MaterialColors#getColor, doesn't it already do that?

**Review by aikrq — APPROVED** [review]

**Comment by trindadedev13** [comment]
I think that after resolving the conflicts this PR can be merged

**Comment by trindadedev13** [comment]
😦

**Review by trindadedev13 — APPROVED** [review]
looks good to me

**Pull request merged**